File: server/router/question/question.go
package question

import (
	"github.com/gin-gonic/gin"
	questionApi "github.com/sxz799/surveyX/api/question"
	"github.com/sxz799/surveyX/middleware"
	questionService "github.com/sxz799/surveyX/service/question"
)

func Question(e *gin.Engine, service *questionService.Service) {
	api := questionApi.NewApi(service)

	g1 := e.Group("/api/admin/question", middleware.JWTAuth())
	{
		g1.GET("/list", api.List)
		g1.POST("", api.Add)
		g1.PUT("", api.Update)
		g1.GET("/:id", api.Get)
		g1.DELETE("/:id", api.Del)
		g1.GET("/analysis/:id", api.Analysis)
	}

	g2 := e.Group("/api/guest/question")
	{
		g2.GET("/list", api.List)
	}
}

File: web/src/api/admin/question.js
import request from '../../utils/request.js'

export function list(data) {
    return request({
        url: '/api/admin/question/list',
        method: 'get',
        params: data
    })
}

export function get(id) {
    return request({
        url: '/api/admin/question/' + id,
        method: 'get',
    })
}

export function add(data) {
    return request({
        url: '/api/admin/question',
        method: 'post',
        data
    })
}

export function update(data) {
    return request({
        url: '/api/admin/question',
        method: 'put',
        data
    })
}


export function del(id) {
    return request({
        url: '/api/admin/question/' + id,
        method: 'delete',
    })
}

export function analysis(id) {
    return request({
        url: '/api/admin/question/analysis/' + id,
        method: 'get',
    })
}
